app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -1,47 +1,261 @@
+                            .icon(bitmapDescriptorFromVector(R.drawable.ic_map_pin))
+                    )
+                }
+        }
+
+        mapViewModel.currentLocation.observe(viewLifecycleOwner) { location ->
+            scooterMap.moveCamera(CameraUpdateFactory.newLatLng(location))
+            mapViewModel.getScooters(location.latitude.toFloat(), location.longitude.toFloat())
+        }
+    }
+
+    private fun bitmapDescriptorFromVector(vectorResId: Int): BitmapDescriptor {
+        val vectorDrawable = ContextCompat.getDrawable(requireContext(), vectorResId)
+        vectorDrawable!!.setBounds(0, 0, vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight())
+        val bitmap = Bitmap.createBitmap(vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight(), Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+        vectorDrawable.draw(canvas)
+        return BitmapDescriptorFactory.fromBitmap(bitmap)
     }
 
-    private fun initListeners() {
-        binding.logoutBtn.setOnClickListener {
-            authViewModel.logOut()
-            findNavController().navigate(MapFragmentDirections.actionMapFragmentToNavigationIntro())
+    private fun initButtons() {
+        requireActivity().onBackPressedDispatcher.addCallback {
+            if (doubleBackPressed) {
+                authViewModel.logOut()
+                requireActivity().finish()
+            } else {
+                doubleBackPressed = true
+                Toast.makeText(requireContext(), getString(R.string.exit_app_button_text), Toast.LENGTH_SHORT).show()
+                Handler().postDelayed({
+                    doubleBackPressed = false
+                }, 3000L)
+            }
         }
+    }
 
-        binding.clearBtn.setOnClickListener {
-            onboardingViewModel.changeLogStatus(logValue = false)
-            authViewModel.logOut()
+    private fun initScooterInfo(markerTitle: String) {
+        mapViewModel.scooterList.value?.forEach { scooter ->
+            if (scooter._id == markerTitle) {
+                binding.apply {
+                    scooterInfo.scooterCV.isVisible = true
+                    scooterInfo.unlockScooterBtn.text = getString(R.string.unlock_scooter_string)
+                    scooterInfo.scooterNumberTV.text = getString(R.string.scooter_number_text, scooter.scooterNumber)
+                    scooterInfo.batteryLevelTV.text = getString(R.string.scooter_battery_level_text, scooter.battery)
+                    scooterInfo.scooterLocationTV.text = getScooterAddress(scooter.location)
+                }
+                setBatteryIcon(scooter.battery)
+            }
+        }
+    }
 
-            findNavController().navigate(MapFragmentDirections.actionMapFragmentToNavigationIntro())
+    private fun setBatteryIcon(batteryLevel: String) {
+        val resource = when (batteryLevel.toInt()) {
+            in 81..100 -> R.drawable.ic_battery_100
```

Comment:
Instead of using `in` you could refactor this method a bit to:
```
    private fun setBatteryIcon(batteryLevel: Int) {
        val resource = when {
            batteryLevel >= 81 -> R.drawable.ic_battery_100
            batteryLevel >= 61 -> R.drawable.ic_battery_80
            // etc ..
            else -> R.drawable.ic_battery_0
        }
        binding.scooterInfo.batteryIndicatorIV.setImageResource(resource)
    }
```